src/bootstrap/native.rs

```diff
@@ -111,6 +116,276 @@ pub fn prebuilt_llvm_config(
     Err(Meta { stamp, build_llvm_config, out_dir, root: root.into() })
 }
 
+pub(crate) fn maybe_download_ci_llvm(builder: &Builder<'_>) {
+    let config = &builder.config;
+    if !config.llvm_from_ci {
+        return;
+    }
+    let mut rev_list = Command::new("git");
+    rev_list.args(&[
+        "rev-list",
+        "--author=bors@rust-lang.org",
+        "-n1",
+        "--first-parent",
+        "HEAD",
+        "--",
+        &format!("{}/src/llvm-project", builder.src.display()),
+        &format!("{}/src/bootstrap/download-ci-llvm-stamp", builder.src.display()),
+        // the LLVM shared object file is named `LLVM-12-rust-{version}-nightly`
+        &format!("{}/src/version", builder.src.display()),
```

Comment:
I'd love to see these moves to builder.src.join(...) -- we should be able to get non-UTF-8 paths correct here pretty easily.